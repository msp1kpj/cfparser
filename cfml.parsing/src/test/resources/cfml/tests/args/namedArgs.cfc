function x(){
var callRate = getServiceLocator().get("object.cps.CallRate");
callRate.init(crid = CallRateId, newName = Name, newDescription = Description);
ArrayAppend(result, callRate);
}