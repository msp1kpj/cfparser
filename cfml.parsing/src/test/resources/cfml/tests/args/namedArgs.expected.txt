/*===TOKENS===*/
FUNCTION                 <function>
IDENTIFIER               <x>
'('                      <(>
')'                      <)>
'{'                      <{>
VAR                      <var>
IDENTIFIER               <callRate>
'='                      <=>
IDENTIFIER               <getServiceLocator>
'('                      <(>
')'                      <)>
'.'                      <.>
IDENTIFIER               <get>
'('                      <(>
OPEN_STRING              <">
STRING_LITERAL           <object.cps.CallRate>
CLOSE_STRING             <">
')'                      <)>
';'                      <;>
IDENTIFIER               <callRate>
'.'                      <.>
IDENTIFIER               <init>
'('                      <(>
IDENTIFIER               <crid>
'='                      <=>
IDENTIFIER               <CallRateId>
','                      <,>
IDENTIFIER               <newName>
'='                      <=>
IDENTIFIER               <Name>
','                      <,>
IDENTIFIER               <newDescription>
'='                      <=>
IDENTIFIER               <Description>
')'                      <)>
';'                      <;>
IDENTIFIER               <ArrayAppend>
'('                      <(>
IDENTIFIER               <result>
','                      <,>
IDENTIFIER               <callRate>
')'                      <)>
';'                      <;>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (element
    (functionDeclaration
      function
      (identifier x)
      (
      parameterList
      )
      (compoundStatement
        {
        (statement
          (localAssignmentExpression
            var
            (multipartIdentifier (identifier callRate))
            =
            (startExpression
              (compareExpression
                (baseExpression
                  (unaryExpression
                    (memberExpression
                      (functionCall
                        (identifierOrReservedWord (identifier getServiceLocator))
                        (
                        argumentList
                        )
                      )
                      .
                      (functionCall
                        (identifierOrReservedWord (identifier get))
                        (
                        (argumentList
                          (argument
                            (startExpression
                              (compareExpression
                                (baseExpression
                                  (unaryExpression
                                    (primaryExpression
                                      (literalExpression (stringLiteral " (stringLiteralPart object.cps.CallRate) "))
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
          ;
        )
        (statement
          (startExpression
            (compareExpression
              (baseExpression
                (unaryExpression
                  (memberExpression
                    (identifier callRate)
                    .
                    (functionCall
                      (identifierOrReservedWord (identifier init))
                      (
                      (argumentList
                        (argument
                          (argumentName (identifier crid))
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression (unaryExpression (memberExpression (identifier CallRateId))))
                            )
                          )
                        )
                        ,
                        (argument
                          (argumentName (identifier newName))
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression (unaryExpression (memberExpression (identifier Name))))
                            )
                          )
                        )
                        ,
                        (argument
                          (argumentName (identifier newDescription))
                          =
                          (startExpression
                            (compareExpression
                              (baseExpression (unaryExpression (memberExpression (identifier Description))))
                            )
                          )
                        )
                      )
                      )
                    )
                  )
                )
              )
            )
          )
          ;
        )
        (statement
          (startExpression
            (compareExpression
              (baseExpression
                (unaryExpression
                  (memberExpression
                    (functionCall
                      (identifierOrReservedWord (identifier ArrayAppend))
                      (
                      (argumentList
                        (argument
                          (startExpression
                            (compareExpression
                              (baseExpression (unaryExpression (memberExpression (identifier result))))
                            )
                          )
                        )
                        ,
                        (argument
                          (startExpression
                            (compareExpression
                              (baseExpression (unaryExpression (memberExpression (identifier callRate))))
                            )
                          )
                        )
                      )
                      )
                    )
                  )
                )
              )
            )
          )
          ;
        )
        }
      )
    )
  )
)
/*======*/